src/core/namespace.c

```diff
@@ -287,12 +288,31 @@ static int apply_mount(
 
                 /* First, get rid of everything that is below if there
                  * is anything... Then, overmount it with an
-                 * inaccessible directory. */
+                 * inaccessible path. */
                 umount_recursive(m->path, 0);
 
-                what = "/run/systemd/inaccessible";
-                break;
+                /* mount() always follow symlinks,
+                 * so we follow symlinks during the stat() as well */
+                r = stat(m->path, &target);
```

Comment:
> sorry, i meant lstat(), of course! sorry for the confusion

That's what I thought, but I was afraid that `fstat()` had some esoteric functionality of which I was not aware of.